static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -0,0 +1,198 @@
+      <UserBadge
+        avatarSize={18}
+        user={dashboardCreator}
+        displayName={
+          <StyledDisplayName>
+            {dashboardCreator?.id === currentUser.id
+              ? `You (${currentUser.email})`
+              : dashboardCreator?.email || currentUser.email}
+          </StyledDisplayName>
+        }
+        displayEmail="Creator"
+      />
+    ),
+    textValue: '_creatorbadge',
+    disabled: dashboardCreator?.id !== currentUser.id,
+    checkboxProps: {
+      isDisabled: true,
+    },
+  });
+
+  // Single team option in the dropdown [WIP]
+  const makeTeamOption = (team: Team) => ({
+    value: team.id,
+    label: `#${team.slug}`,
+    leadingItems: <TeamAvatar team={team} size={18} />,
+  });
+
+  // Avatars/Badges in the Edit Selector Button
+  const triggerAvatars =
+    isEverythingSelected || !dashboardCreator ? (
+      <StyledBadge text={'All'} />
+    ) : (
+      <StyledAvatarList key="avatar-list" users={[dashboardCreator]} avatarSize={25} />
+    );
+
+  const dropdownOptions = [
+    makeCreatorOption(),
+    {
+      value: '_everyone_section',
+      options: [
+        {
+          value: '_everyone',
+          label: 'Everyone',
+          disabled: dashboardCreator?.id !== currentUser.id,
+        },
+      ],
+    },
+    // [WIP: Selective edit access to teams]
+    {
+      value: '_teams',
+      label: t('Teams'),
+      options: teams.map(makeTeamOption),
+      showToggleAllButton: true,
+      disabled: true,
+    },
+  ];
+
+  // Handles state change when dropdown options are selected
+  const onSelectOptions = newSelectedOptions => {
+    const newSelectedValues = newSelectedOptions.map(
+      (option: {value: string}) => option.value
+    );
+    isEverythingSelected = false;
+    if (newSelectedValues.includes('_everyone')) {
+      isEverythingSelected = true;
+      setselectedOptions(['_everyone', '_creator', ...teamIds]);
+      if (newDashboardPermissions) {
+        newDashboardPermissions.isCreatorOnlyEditable = false;
+      }
+    } else if (!newSelectedValues.includes('_everyone')) {
+      setselectedOptions(['_creator']);
+      if (newDashboardPermissions === undefined) {
+        // When the dashboard does not have a permissions model associated
+        setNewDashboardPermissions({isCreatorOnlyEditable: true});
+      } else {
+        newDashboardPermissions.isCreatorOnlyEditable = true;
+      }
+    }
+  };
+
+  const dropdownMenu = (
+    <StyledCompactSelect
+      size="sm"
+      onChange={newSelectedOptions => {
+        onSelectOptions(newSelectedOptions);
+      }}
+      onClose={() => {
+        if (!isEqual(newDashboardPermissions, dashboard.permissions)) {
+          onChangeEditAccess?.(newDashboardPermissions);
+        }
+      }}
+      multiple
+      searchable
+      options={dropdownOptions}
+      value={selectedOptions}
+      triggerLabel={[
+        <React.Fragment key="edit-access-label">{t('Edit Access:')}</React.Fragment>,
```

Comment:
Typically we import `Fragment` directly from `react` so we don't do `React.Fragment`. Also, `Fragment` is typically used to wrap multiple elements that belong together but don't exactly have a root, since they're just wrapping a single thing, you should be able to do `triggerLabel={[t('Edit Access:'), triggerAvatars]}`